src/components/Tx/SendTx/CurrencySelect.jsx

```diff
@@ -0,0 +1,223 @@
+import React, { Component } from 'react'
+import PropTypes from 'prop-types'
+import styled from 'styled-components'
+import Identicon from '../../Identicon'
+
+export default class CurrencySelect extends Component {
+  static displayName = 'CurrencySelect'
+
+  static propTypes = {
+    etherWallet: PropTypes.shape({
+      balance: PropTypes.string,
+      symbol: PropTypes.string,
+      address: PropTypes.string,
+      name: PropTypes.string
+    }),
+    onSelect: PropTypes.func,
+    tokens: PropTypes.arrayOf(
+      PropTypes.shape({
+        balance: PropTypes.string,
+        symbol: PropTypes.string,
+        address: PropTypes.string,
+        name: PropTypes.string
+      })
+    )
+  }
+
+  static defaultProps = {
+    etherWallet: {},
+    tokens: []
+  }
+
+  chooseCurrency = event => {
+    const { onSelect } = this.props
+    if (onSelect) {
+      onSelect(event)
+    }
+  }
+
+  static renderCurrencySpans = currency => {
```

Comment:
Would prefer just to leave this a class method (not `static`). Its usage is uncommon/unexpected within the render function: ComponentName.funcName()